src/latexify/codegen/function_codegen.py

```diff
@@ -558,7 +558,28 @@ def _get_sum_prod_range(self, node: ast.comprehension) -> tuple[str, str] | None
             lower_rhs = f"{{{range_info.start_int}}}"
 
         if range_info.stop_int is None:
-            upper = "{" + self.visit(range_info.stop) + " - 1}"
+            # special processing needed if range_info.stop involves addition or subtraction
+            if (
+                isinstance(range_info.stop, ast.BinOp)
+                and isinstance(range_info.stop.op, (ast.Add, ast.Sub))
+                and isinstance(range_info.stop.right, ast.Constant)
```

Comment:
`ast.Constant` is only available from Python 3.8. in 3.7 this logic should use `ast.Num`

Reply:
That means we need to implement 2 variants of this logic, which are switched by `if sys.version_info.minor < 8`